vmas/scenarios/navigation.py

```diff
@@ -24,8 +24,16 @@ def make_world(self, batch_dim: int, device: torch.device, **kwargs):
         self.n_agents = kwargs.pop("n_agents", 4)
         self.collisions = kwargs.pop("collisions", True)
 
-        self.x_semidim = kwargs.pop("x_semidim", None)
-        self.y_semidim = kwargs.pop("y_semidim", None)
+        self.world_spawning_x = kwargs.pop("world_spawning_x", 1)
+        self.world_spawning_y = kwargs.pop("world_spawning_y", 1)
+        self.enforce_bounds = kwargs.pop("enforce_bounds", False)
+
+        if self.enforce_bounds:
+            self.x_semidim = self.world_spawning_x
+            self.y_semidim = self.world_spawning_y
+        else:
+            self.x_semidim = None
+            self.y_semidim = None
```

Comment:
Could we move this after the kwargs reading?

Also maybe a small comment after each of the 3 new kwargs would help

Reply:
Yes, it definitely helps. I tried to fix this in my latest commit :)